File: frontend/src/api/translation.ts
import client from './client'
import type { Language } from '../i18n/I18nProvider'

interface TranslateRequest {
  text: string
  target_language: Language
  source_language?: Language | 'unknown'
}

interface TranslateResponse {
  text: string
  detected_language: string
  translated: boolean
}

interface TranslateBatchRequest {
  items: string[]
  target_language: Language
  source_language?: Language | 'unknown'
}

interface TranslateBatchResponse {
  items: string[]
  detected_language: string
  translated_count: number
}

export async function translateDescription(payload: TranslateRequest): Promise<TranslateResponse> {
  const response = await client.post<TranslateResponse>('/translate/description', payload)
  return response.data
}

export async function translateBatch(payload: TranslateBatchRequest): Promise<TranslateBatchResponse> {
  const response = await client.post<TranslateBatchResponse>('/translate/batch', payload)
  return response.data
}

File: backend/app/schemas/translation.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class TranslateDescriptionRequest(BaseModel):
    text: str = Field(min_length=1)
    target_language: Literal['en', 'fr']
    source_language: Literal['en', 'fr', 'unknown'] = 'unknown'


class TranslateDescriptionResponse(BaseModel):
    text: str
    detected_language: Literal['en', 'fr', 'unknown']
    translated: bool


class TranslateBatchRequest(BaseModel):
    items: list[str] = Field(min_length=1)
    target_language: Literal['en', 'fr']
    source_language: Literal['en', 'fr', 'unknown'] = 'unknown'


class TranslateBatchResponse(BaseModel):
    items: list[str]
    detected_language: Literal['en', 'fr', 'unknown']
    translated_count: int
